sympy/stats/symbolic_probability.py

```diff
@@ -158,7 +217,8 @@ def _eval_rewrite_as_Probability(self, arg, condition=None, **kwargs):
                 return Sum(arg.replace(rv, symbol)*Probability(Eq(rv, symbol), condition), (symbol, rv.pspace.domain.set.inf, rv.pspace.set.sup))
 
     def _eval_rewrite_as_Integral(self, arg, condition=None, **kwargs):
-        return expectation(arg, condition=condition, evaluate=False)
+        return self.func(arg, condition=condition,
+                evaluate=False).doit(deep=False, for_integral=True)
```

Comment:
Are you also handling the case of discrete random variables? They should be rewritten as `Sum`.

Reply:
Yes, added a docstring for the same